### ISSUE TITLE
Investigate Google offerings for static site hosting

### ISSUE THREAD

**Issue by pmac** [mozmeao/infra#1037]

**Comment by pmac** [comment]
Looks like Firebase will do most of the things that we need and that we have working at Netlify for the hosted site:

https://firebase.google.com/docs/hosting/full-config

The primary difference is that Netlify will hook directly to Github and build things for us, while Google Firebase relies on installing a NodeJS CLI tool and deploying with that. I'm sure that build and deployment could happen in our own CI, but the Netlify method is much more convenient for us.

**Comment by pmac** [comment]
But if it's that much easier to pay Google than anyone else then perhaps it's our best static hosting option for now.

@metadave thoughts?

**Comment by metadave** [comment]
some thoughts:

- is "Netlify convenience" a matter of hours/days/weeks for the equivalent in Firebase?
- can we use a standardized Docker image to do the pushes to Firebase across all of our static websites?
- how long would it take to get careers.m.o working on Firebase? 
  - would we have issues [switching certs](https://github.com/mozmeao/lumbergh/issues/369)? 

The Netlify CASA ticket may take some time to go through, it may be beneficial to look into Firebase during that process to make sure we're choosing the right hosting company.

**Comment by pmac** [comment]
> is "Netlify convenience" a matter of hours/days/weeks for the equivalent in Firebase?

Days I'd say. It's a matter of setting up CI for Firebase. I'm not sure what secrets and auth for that would entail, but I'm sure it's doable.

> * can we use a standardized Docker image to do the pushes to Firebase across all of our static websites?

That's certainly plausible.
 
> * how long would it take to get careers.m.o working on Firebase?

I wouldn't imagine it would take long at all. Automating updates would be the most time consuming. Right now @glogiotatidis has it setup so that updates in Greenhouse ping Netlify and cause a rebuild. We could have Greenhouse ping GitLab instead, but I'm not sure how much effort that would be to setup.

>   * would we have issues [switching certs](https://github.com/mozmeao/lumbergh/issues/369)?

We might well have cert issues. Google [claims it can take a few days](https://firebase.google.com/docs/hosting/custom-domain#wait_for_ssl_certificate_provisioning). I'll look to see if there's manual cert upload as well so that we can do a similar thing to what was done with Netlify.

**Comment by metadave** [comment]
Gitlab provides hooks as well:
https://docs.gitlab.com/ee/ci/triggers/#triggering-a-pipeline-from-a-webhook

note, the [certs](https://firebase.google.com/docs/hosting/custom-domain#wait_for_ssl_certificate_provisioning) link above mentioned several hours, not days (unless I'm missing something).

**Comment by pmac** [comment]
Update on certs, that doc I linked above under the "setup your domain for hosting" section in step 5 it says that you can do a zero-downtime migration. Basically the procedure is:

1. they give you a value to add to a TXT record in DNS for the ACME protocol
2. they use that to automatically provision a cert for you and install it
3. once the above is done you can point DNS to firebase and all should be well

I created a test site and verified my domain and it gave me these options under "advanced" when setting up the custom domain.

**Comment by pmac** [comment]
You're right. up to 24 hours... so I figured a couple of "work" days... but you're right. And my above comment means that it doesn't much matter anyway.

**Comment by metadave** [comment]
It would certainly be nice to have our stuff [mostly](https://images.gr-assets.com/hostedimages/1486305408ra/21902383.gif) in one place (GCP), vs 3 cloud vendors.

**Comment by bensternthal** [comment]
One thing to note about the certs....

These do use letsencrypt and the certs cover multiple domains. See screenshot for my personal website (devpatch.com). For me this was funky at first before I realized what was happening. 

I was like.. what is this weirdo domain .. has google given me the wrong cert?

<img width="485" alt="screen shot 2019-02-26 at 8 19 41 am" src="https://user-images.githubusercontent.com/198507/53428284-5d55d400-399f-11e9-8292-708f08a611bd.png">
<img width="481" alt="screen shot 2019-02-26 at 8 17 31 am" src="https://user-images.githubusercontent.com/198507/53428285-5d55d400-399f-11e9-8c93-1a319dbc8be2.png">

**Comment by metadave** [comment]
wow, that's an ugly cert.
